### PR TITLE
Support Windows workers for docker builds

### PR DESCRIPTION
**Pull request by cachedout**
## What does this PR do?

Catches a condition where sometimes workers labled as `docker` can be Windows.

## Why is it important?
Corrects occasional build failures such as this one: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-docker-opbeans-pipeline/detail/apm-docker-opbeans-pipeline/779/pipeline

### CODE REVIEW

File: .ci/dockerImagesOpbeans.groovy [hunk 1]
```diff
@@ -182,10 +182,18 @@ def buildDockerImage(args){
       withEnv(env){
         retry(3) {
           sleep randomNumber(min: 5, max: 10)
-          sh(label: "build docker image", script: "docker build ${options} -t ${image} .")
+          if(isUnix()){
+            sh(label: "build docker image", script: "docker build ${options} -t ${image} .")
+          } else {
+            bat(label: "build docker image", script: "docker build ${options} -t ${image} .")
+          }
```
Comment (v1v) [inline .ci/dockerImagesOpbeans.groovy:189]: You can use https://github.com/elastic/apm-pipeline-library/blob/master/vars/cmd.groovy

```suggestion
          cmd(label: "build docker image", script: "docker build ${options} -t ${image} .")
```

File: .ci/dockerImagesOpbeans.groovy [hunk 2]
```diff
@@ -182,10 +182,18 @@ def buildDockerImage(args){
...
         }
         if(push){
-          sh(label: "push docker image", script: "docker push ${image}")
+          if(isUnix()){
+            sh(label: "push docker image", script: "docker push ${image}")
+          } else {
+            bat(label: "push docker image", script: "docker push ${image}")
+          }
```
Comment (v1v) [inline .ci/dockerImagesOpbeans.groovy:196]:
```suggestion
          cmd(label: "push docker image", script: "docker push ${image}")
```

**Review by v1v — COMMENTED** [review]
`cmd` could be a good candidate

File: .ci/dockerImagesOpbeans.groovy [hunk 3]
```diff
@@ -182,10 +182,18 @@ def buildDockerImage(args){
       withEnv(env){
         retry(3) {
           sleep randomNumber(min: 5, max: 10)
-          sh(label: "build docker image", script: "docker build ${options} -t ${image} .")
+          if(isUnix()){
+            sh(label: "build docker image", script: "docker build ${options} -t ${image} .")
```
Comment (kuisathaverat) [inline .ci/dockerImagesOpbeans.groovy:186]: cmd make the trick if the command is the same

https://github.com/elastic/apm-pipeline-library/blob/master/vars/README.md#cmd

**Review by v1v — APPROVED** [review]

**Pull request merged**